CI: pkg version check fails on forks due to upstream tag dependency

### Bug Description

The CI checks fail on the fork with the following error:
```
These packages have changed since v1.10.0 and need a version increment:
     ./base/all ./base/db ./base/logger ./base/qaqc ./base/remote ./base/settings ./base/utils ./base/visualization ./base/workflow ./models/basgra ./models/biocro ./models/cable ./models/clm45 ./models/dalec ./models/dvmdostem ./models/ed ./models/fates ./models/gday ./models/jules ./models/ldndc ./models/linkages ./models/lpjguess ./models/maat ./models/maespa ./models/preles ./models/rothc ./models/sibcasa ./models/sipnet ./models/stics ./models/template ./modules/allometry ./modules/assim.batch ./modules/assim.sequential ./modules/benchmark ./modules/data.atmosphere ./modules/data.land ./modules/data.mining ./modules/data.remote ./modules/emulator ./modules/meta.analysis ./modules/photosynthesis ./modules/priors ./modules/rtm ./modules/uncertainty ./shiny/BenchmarkReport ./shiny/Data-Ingest ./shiny/dbsync ./shiny/Elicitation ./shiny/global-sensitivity ./shiny/Pecan.depend ./shiny/ViewMet ./shiny/workflowPlot
Edit the DESCRIPTION file(s) to add '.9000' to the 'Version:' field.
Error: Process completed with exit code 1.
```

<img width="663" height="266" alt="Image" src="https://github.com/user-attachments/assets/c177d8e4-494d-41ff-9ae3-698aaa0a5bee" />
 

Should we address this, even though it will not affect anything upstream? Maybe, we can skip this check in the fork, or fetch the upstream tag within the fork to resolve the issue?

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -72,5 +72,5 @@ jobs:
         tag=$(\
           curl -s https://api.github.com/repos/pecanproject/pecan/releases/latest \
           | sed -n 's!.*"tag_name": "\(.*\)".*!\1!p')
-        git fetch --tags --depth=1
+        git fetch https://github.com/PecanProject/pecan.git "refs/tags/${tag}:refs/tags/${tag}" --depth=1
         ./scripts/ensure_version_bump.sh "$tag"

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -70,7 +70,7 @@ jobs:
     - name: check package versions
       run: |
         tag=$(\
-          curl -s https://api.github.com/repos/pecanproject/pecan/releases/latest \
+          curl -s https://api.github.com/repos/PecanProject/pecan/releases/latest \
           | sed -n 's!.*"tag_name": "\(.*\)".*!\1!p')
         git fetch https://github.com/PecanProject/pecan.git "refs/tags/${tag}:refs/tags/${tag}" --depth=1
         ./scripts/ensure_version_bump.sh "$tag"

